index.js

```diff
@@ -58,6 +63,28 @@ exports.getPostThread = function (ssb, mid, opts, cb) {
   })
 }
 
+exports.reviseFlatThread = function(ssb, thread, callback) {
+  // function that calls getLatestRevision on each of the elements of a flat
+  // thread, and collects them up into one callback.
+  var callbackAggregator = multicb({pluck: 1})
+
+  thread.forEach(function (thisMsg) { 
+    // we don't *need* to get the latest revision of an edit, it's included
+    // already
+    if (thisMsg.value.content.type === 'post') {
+      exports.getLatestRevision(ssb, thisMsg, callbackAggregator())
+    }
+  })
+  
+  callbackAggregator(function (err, results) {
+    if (err) {
+      callback(err)
+    } else {
+      callback(null, results)
+    }
+  })
+}
```

Comment:
i think it does so deliberately. the idea is that anything that is a post has a latest revision (case 1), and anything that is a post-edit is either the latest revision of some post (case 2), or it isn't (case 3).

- case 1: replace with latest rev
- case 2: will be handled by case 1
- case 3: would not be displayed anyway